### PR TITLE
#4203 Set vaccinator as prescriber for dispensing history modal if available

### LINKED ISSUE #4203: Show vaccinator in patient history for online patient lookups
## Is your feature request related to a problem? Please describe.

**A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]**

For Tonga we added a feature to allow patient history lookup for all patients in the vaccine module (limited to vaccination history only): https://github.com/openmsupply/mobile/issues/4022

Tonga/BES have now asked that they would also like to be able to see that vaccinator/prescriber in the patient history for patients that have been looked up online.

This may require a change to the 4D api as vaccinator is probably not currently provided by the api.

**A clear and concise description of what you want to happen.**

Currently when you look up a patient who was vaccinated on another tablet in the dispensing module you can see the item, quantity and date of the vaccine in the patient history. But currently prescriber (vaccinator) shows N/A. Tonga would like to see who the vaccinator was.

## Implementation

**A description of how the solution would best be implemented in the code base**

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

## Additional context

**Add any other context or screenshots about the feature request here.**

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #4203

## Change summary

- Small change to set the vaccinator as the prescriber for the Dispensing History modal if available when fetching remote history records (since this modal only displays a prescriber column)

image

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- Prerequisites:
  - Run `z_toggleDangerousPatientMode` (⚠️!)  on mSupply Desktop
  - Create a vaccination on a store (ensure it has a vaccinator selected)
  - Login to a different store on mobile and navigate to Dispensing -> History

- [ ] Confirm that the vaccine history record shows up, and the vaccinator is displayed

### Related areas to think about
Not sure if it is super clear that a prescriber/vaccinator are interchangeable on this page.
Some possible ways to make it clearer (but weren't done as I wasn't sure...):
- Adding an extra 'vaccinator' column (but probably too much clutter).
- Changing the column heading to 'Prescriber/Vaccinator' if vaccine module is enabled?

### CODE REVIEW

File: src/sync/lookupApiUtils.js [hunk 1]
```diff
@@ -237,7 +237,7 @@ export const getPatientHistoryResponseProcessor = ({
           doses: totalQuantity * doses,
           itemCode,
           itemName,
-          prescriber,
+          prescriber: !isVaccine && vaccinator ? vaccinator : prescriber,
```
Comment (mark-prins) [inline src/sync/lookupApiUtils.js:240]: shouldn't that be `isVaccine && vaccinator`?

Comment (katherine-sussol) [reply src/sync/lookupApiUtils.js:240]: 🤣  I considered renaming `isVaccine` - maybe I should have! It looks like it's used to determine whether we are on the Dispensing history or the Vaccine Dispensing history? So `!isVaccine` actually means we are on Dispensing History

Comment (mark-prins) [reply src/sync/lookupApiUtils.js]: oh 🤦  I see now. thanks.

**Comment by mark-prins** [comment]
I'm still concerned at the naming of the column, which is why I left the vaccinator out originally.
I thought there would be confusion between prescriber and vaccinator and didn't like displaying the vaccinator in a column headed 'PRESCRIBER'. Too much to show 'PRESCRIBER / VACCINATOR'?

**Review by mark-prins — DISMISSED** [review]
Code is fine - as noted, I am troubled by the ambiguous column title. If others are fine with that then that's all good.

**Comment by katherine-sussol** [comment]
> Code is fine - as noted, I am troubled by the ambiguous column title. If others are fine with that then that's all good.

Thanks @mark-prins - sounds like the troubling is mutual. Have discussed with Nick - will update the implementation to show PRESCRIBER / VACCINATOR if vaccine dispensing is enabled (i.e. Dispensary is enabled & Vaccine Stock is available), otherwise leave as PRESCRIBER (otherwise it may be confusing if the app otherwise doesn't have much of a concept of 'Vaccinations').

**Comment by katherine-sussol** [comment]
The header on the dispensing history modal has been updated 15e4eef.
Re-requesting review just as I've changed a bit of stuff around (have created a separate modal definition for when a remote history record has a vaccinator) 😅

**Review by joshxg — DISMISSED** [review]
Working well

Getting a blank cell when first opening- I added some getters to `TransactionBatch` - unsure if it's worthwhile.

```
get prescriber(){
...
}

get prescriberOrVaccinator(){ 
    return this.vaccinator || this.prescriber
}
```

**Comment by joshxg** [comment]
Oh and this is based on master - I don't know the priority etc so unsure if that's correct

**Comment by katherine-sussol** [comment]
Oh, I didn't mean to dismiss the last review. Thanks though I think it looks a lot better, keep forgetting about realm getters 😸 
aafc3795

**Comment by katherine-sussol** [comment]
> Oh and this is based on master - I don't know the priority etc so unsure if that's correct

Oh hmmm. It sounded like Tonga were wanting it asap - but I don't know if asap is sooner than Kiribati want their changes...

**Comment by joshxg** [comment]
Yeah, I prob `8.1.2`

**Review by mark-prins — APPROVED** [review]
Yup - that looks good to me :)

I have some funky behaviour still with the history returning and changing the sort order or populating the vaccinator or, and this is the best one, _removing_ the vaccinator ( was vaccinated on this tablet, but not sync'd to server )

That's all unrelated to this PR though, and issues which we should address another day.

**Comment by mark-prins** [comment]
> Yeah, I prob `8.1.2`

great! `8.2.0` it is :)

**Comment by ArpitaSussol** [comment]
@katherine-sussol can you please provide me z_toggleDangerousPatientMode method to run FC?

**Comment by katherine-sussol** [comment]
> @katherine-sussol can you please provide me z_toggleDangerousPatientMode method to run FC?

I've sent it via Telegram - let me know if any issues!

**Pull request merged**